src/utils/Config.ts

```diff
@@ -0,0 +1,18 @@
+import * as dotenv from 'dotenv';
+
+export const loadConfig = () => {
```

Comment:
I have very good experience with https://www.npmjs.com/package/config. It by default supports files per environment (in many formats including JSON and YAML) and it has support for environment variables as possible way to override defaults.

In terms of choice I guess it comes down to personal preference.

If you wanna stick with `dotenv` please simplify config loading. You can basically validate environment type using regex for example and then construct use the value to dynamically evaluate path e.g. `path = '${__dirname}/../../.env.${process.env.NODE_ENV}'`. Having some many switch cases will make maintenance more difficult.

Reply:
https://www.npmjs.com/package/dotenv#what-happens-to-environment-variables-that-were-already-set <- @n-r-b , @akaJSaunders worth to have a read if we stick to `dotenv`.

Reply:
Let me know which library you prefer for this, I don't mind either but I've used dotenv before

Reply:
Lets stick to `dotenv` since you started. As long as there is easy way to configure service via environment variables I am happy.